integration-tests/google-pubsub/src/test/java/org/apache/camel/quarkus/component/google/pubsub/it/GooglePubSubCustomizer.java

```diff
@@ -0,0 +1,213 @@
+
+            final String groupedSubscriptionName = "camel-quarkus-grouped-subscription-"
+                    + RandomStringUtils.randomAlphanumeric(49).toLowerCase(Locale.ROOT);
+            envContext.property("google-pubsub.grouped-subscription-name", groupedSubscriptionName);
+
+            final String orderingTopicName = "camel-quarkus-ordering-topic-"
+                    + RandomStringUtils.randomAlphanumeric(49).toLowerCase(Locale.ROOT);
+            envContext.property("google-pubsub.ordering-topic-name", orderingTopicName);
+
+            final String orderingSubscriptionName = "camel-quarkus-ordering-subscription-"
+                    + RandomStringUtils.randomAlphanumeric(49).toLowerCase(Locale.ROOT);
+            envContext.property("google-pubsub.ordering-subscription-name", orderingSubscriptionName);
+
+            final String ackTopicName = "camel-quarkus-ack-topic-"
+                    + RandomStringUtils.randomAlphanumeric(49).toLowerCase(Locale.ROOT);
+            envContext.property("google-pubsub.ack-topic-name", ackTopicName);
+            final String ackSubscriptionName = "camel-quarkus-ack-subscription-"
+                    + RandomStringUtils.randomAlphanumeric(49).toLowerCase(Locale.ROOT);
+            envContext.property("google-pubsub.ack-subscription-name", ackSubscriptionName);
+
+            Topic topic = createTopic(topicClient, topicName, projectId);
+            Subscription subscription = createSubscription(subscriptionClient, topic, subscriptionName, projectId);
+            Topic topicGrouped = createTopic(topicClient, groupedTopicName, projectId);
+            Subscription subscriptionGrouped = createSubscription(subscriptionClient, topicGrouped,
+                    groupedSubscriptionName, projectId);
+            Topic topicOrdering = createTopic(topicClient, orderingTopicName, projectId);
+            Subscription subscriptionOrdering = createSubscription(subscriptionClient, topicOrdering,
+                    orderingSubscriptionName, projectId, true);
+            Topic topicAck = createTopic(topicClient, ackTopicName, projectId);
+            Subscription subscriptionAck = createSubscription(subscriptionClient, topicAck,
+                    ackSubscriptionName, projectId);
+
+            envContext.closeable(() -> {
+                subscriptionClient.deleteSubscription(subscriptionGrouped.getName());
+                topicClient.deleteTopic(topicGrouped.getName());
+                subscriptionClient.deleteSubscription(subscriptionAck.getName());
+                topicClient.deleteTopic(topicAck.getName());
+                subscriptionClient.deleteSubscription(subscriptionOrdering.getName());
+                topicClient.deleteTopic(topicOrdering.getName());
+                subscriptionClient.deleteSubscription(subscription.getName());
+                topicClient.deleteTopic(topic.getName());
+
+                topicClient.shutdown();
```

Comment:
Do you think it would work to `shutdown()` both and then `awaitTermination()` both both at the same time ? Would it be faster anyway ?

Reply:
You are right, I'll fix it.

Reply:
done

Reply:
Great that we do this :+1: